digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/service/FileSyncService.java

```diff
@@ -0,0 +1,136 @@
+import org.digit.health.sync.kafka.Producer;
+import org.digit.health.sync.service.checksum.Checksum;
+import org.digit.health.sync.service.checksum.MD5Checksum;
+import org.digit.health.sync.service.compressor.Compressor;
+import org.digit.health.sync.service.compressor.GzipCompressor;
+import org.digit.health.sync.web.models.AuditDetails;
+import org.digit.health.sync.web.models.FileDetails;
+import org.digit.health.sync.web.models.ReferenceId;
+import org.digit.health.sync.web.models.SyncErrorDetailsLog;
+import org.digit.health.sync.web.models.SyncLog;
+import org.digit.health.sync.web.models.SyncStatus;
+import org.digit.health.sync.web.models.request.SyncUpDto;
+import org.egov.common.contract.request.User;
+import org.egov.tracer.model.CustomException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpServerErrorException;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.UUID;
+
+@Slf4j
+@Service
+public class FileSyncService implements SyncService {
+
+    private final Producer producer;
+    private final ObjectMapper objectMapper;
+    private final FileStoreService fileStoreService;
+    private final Compressor compressor;
+    private final Checksum checksum;
+
+
+    @Autowired
+    public FileSyncService(Producer producer, FileStoreService fileStoreService, ObjectMapper objectMapper, GzipCompressor compressor, MD5Checksum checksumValidator) {
+        this.producer = producer;
+        this.fileStoreService = fileStoreService;
+        this.objectMapper = objectMapper;
+        this.compressor = compressor;
+        this.checksum = checksumValidator;
+    }
+
+    @Override
+    public String sync(SyncUpDto syncUpDto) {
+        String tenantId = syncUpDto.getRequestInfo().getUserInfo().getTenantId();
+        SyncLog syncLog = initiateSyncLog(syncUpDto);
+        FileDetails fileDetails = syncUpDto.getFileDetails();
+        try {
+            byte[] data = fileStoreService.getFile(fileDetails.getFileStoreId(), tenantId);
+            checksum.validate(data, fileDetails.getChecksum());
+            HashMap json = objectMapper.readValue(read(data), HashMap.class);
+            log.info(json.entrySet().toString());
+            persistSyncLog(syncLog);
+        } catch (JsonProcessingException ex) {
+            handleSyncError(syncLog, ex, "Invalid JSON");
+        } catch (HttpServerErrorException | IOException ex) {
+            handleSyncError(syncLog, ex, "Invalid File");
+        } catch (Exception ex) {
+            handleSyncError(syncLog, ex, "Unable to process");
+        }
+        return syncLog.getSyncId();
+
+    }
+
+
+    private String read(byte[] data) throws IOException {
+        BufferedReader br = compressor.decompress(new ByteArrayInputStream(data));
```

Comment:
Refactor decompress to take and receive byte array

Reply:
Done